core/deployment/src/main/java/io/quarkus/deployment/configuration/ConfigDefinition.java

```diff
@@ -58,6 +59,10 @@
 
     public static final String NO_CONTAINING_NAME = "<<ignored>>";
 
+    private static final String FIRST_CONFIG_SEGMENT = "quarkus";
+    private static final List<String> FALSE_POSITIVE_QUARKUS_CONFIG_MISSES = Collections
```

Comment:
Not sure about the name. But doing that would allow us to be more consistent as right now, we have a couple of system properties for which we used a dash (`quarkus-...`). I think being consistent would be better.

Reply:
Not sure I follow exactly. Would you like me to also ignore errors for properties starting with `quarkus-`?

Reply:
Or perhaps would you like me to just change the name of field?

Reply:
Sorry if it was unclear. Yeah so I'm not sure the name is that good but it's hard to find a better one.

The other (separate) thing I mentioned is that we had a tendency to add system properties of this form `quarkus-` to avoid getting warnings (at least, that's my understanding of it). If we start an ignore list, we could make them `quarkus.` and fully consistent with the rest of our config.

@stuartwdouglas would know more as he was the one introducing these system properties. Maybe he chose `quarkus-` intentionally.

Reply:
Cool, thanks for the explanation. So I left it as is for now but can of course change if we come up with a better name or some more properties that should be here.

Reply:
Searching naively through the codebase I didn't find any `quarkus-` system props :thinking:

Reply:
But there uses to be `private static final String RESOURCES_PROP = "quarkus-internal.undertow.resources";` in `DevMojo`, see: https://github.com/quarkusio/quarkus/blob/88c87731a3e6a815d3499b2c4fa495e5da539d64/devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

Reply:
So @gsmet unless you have found some other code, I think we don't have to worry about `quarkus-` anymore :)

Reply:
@geoand you're right, it has been changed to `quarkus.debug.generated-classes-dir`. I don't know if it's ignored right now or if we should add it too to the ignore list.

Reply:
I haven't seen log output complain about it, have you?

Reply:
Ah, you are right! I just spotted it in the output. Updating the PR now.

Reply:
PR updated

Reply:
@gsmet ok to merge?